python/tank/platform/validation.py

```diff
@@ -379,16 +389,31 @@ def validate(self, settings):
         # first sanity check that the schema is correct
         validate_schema(self._display_name, self._schema)
         
-        # Ensure that all required keys are in the settings and that the
-        # values are appropriate.
+        # Ensure that all keys are in the settings or have a default value in
+        # the manifest. Also make sure values are appropriate.
         for settings_key in self._schema:
             value_schema = self._schema.get(settings_key, {})
-            
-            # make sure the required key exists in the environment settings
-            if settings_key not in settings:
-                raise TankError("Missing required key '%s' in settings!" % settings_key)
-            
-            self.__validate_settings_value(settings_key, value_schema, settings[settings_key])
+
+            if settings_key in settings:
+                # value exists in the settings. use it.
+                settings_value = settings[settings_key]
+            else:
+                # No value in the settings. Look for a default value.
+                if value_schema.get("type") == "hook":
+                    # Hooks already have the concept of a default value. If this
+                    # setting is a hook, just use that special default hook
+                    # string and allow the validation code do its thing.
+                    settings_value = constants.TANK_BUNDLE_DEFAULT_HOOK_SETTING
```

Comment:
If we use the value from the manifest, we'll need to modify all the manifests to use the "{self}" form instead of just the hook name. I tried to fall back directly to the manifest originally, and the apps failed because they were looking for hooks in the config's `hooks` directory since the legacy defaulting of the path wasn't kicking in. It would be good to get @manneohrstrom to have a look here as well.